shared/dataflow/codeql/dataflow/internal/DataFlowImpl.qll

```diff
@@ -1328,6 +1331,15 @@ module MakeImpl<InputSig Lang> {
           fwdFlow1(_, _, _, _, _, _, t0, t, ap, _) and t0 != t
         }
 
+        bindingset[c, t, tail]
+        additional Ap apCons(Content c, Typ t, Ap tail) {
+          result = Param::apCons(c, t, tail) and
+          exists(int limit |
+            limit = Config::accessPathLimit() and
+            if tail instanceof ApNil then limit > 0 else limit > 1
+          )
```

Comment:
How does this affect DIL/RA? Does it compile away inder normal circumstances?

Reply:
Actually, you can move the `limit > 0` check into the `useFieldFlow` predicate. Then you get filtering already in stage 1.

Reply:
... after which you can assume `limit > 0`, so no need to tweak the `ConsNil` case in `AccessPathApprox` after all.

Reply:
Sadly, no, but the optimizer should be able to. The added DIL is
```
        (
          (
            `DataFlowImpl::Impl<SqlInjectionQuery::SqlInjectionFlow::C>::Stage2Param::ApNil#ec757912`(ap0) and
            GT(5, 0)
          )
          or
          (
            not(
              `DataFlowImpl::Impl<SqlInjectionQuery::SqlInjectionFlow::C>::Stage2Param::ApNil#ec757912`(ap0)
            ) and
            GT(5, 1)
          )
        )

```

Reply:
And you really only need the `limit > 1` check in the `Stage2Param` version of `apCons` - subsequent stages will then get the condition by invariant.